app/rww/ldp/WebClient.scala

```diff
@@ -154,16 +151,19 @@ class WSClient[Rdf <: RDF](readerSelector: ReaderSelector[Rdf, Try], rdfWriter:
     val maybeMimeType = maybeContentType.flatMap(ct => MimeType.parse(ct))
     maybeMimeType.map {
       _ match {
-        case MimeType("text", "plain", _) => AllRdfMimeTypes
-        case MimeType("application", "xml", _) => Set(MimeType("application", "xml"), MimeType("application", "rdf+xml"))
-        case MimeType("text", "xml", _) => Set(MimeType("text", "xml"), MimeType("application", "rdf+xml"))
-        case mimeT => Set(mimeT)
+        case MimeType("text", "plain", _)      => AllRdfMimeTypes
+        case MimeType("application", "xml", _) => Set(MimeType("application", "xml"), MimeType("application", "rdf+xml"), MimeType("text", "rdf+xml"))
+        case MimeType("text", "xml", _)        => Set(MimeType("text", "xml"), MimeType("application", "rdf+xml"), MimeType("text", "rdf+xml"))
+        case MimeType(_, "rdf+xml", _)         => Set(MimeType("application", "rdf+xml"), MimeType("text", "rdf+xml")) //banana-rdf has RdfXml mime type specified as MimeType("text", "rdf+xml") instead of MimeType("application", "rdf+xml")
+        case MimeType("text", "n3", _)         => Set(MimeType("application", "ntriples"), MimeType("text", "n3"))  //banana-rdf has NTriples mime type specified as MimeType("application", "ntriples") instead of MimeType("text", "n3")
```

Comment:
text/n3 is the most advanced syntax for which there are few implementations. What one should hope is that turtle could parse it. it's very unlikely that ntriples can parse it. So in all these cases I'd change the `Set` to a `List`  and put first the actually requested mime type ( in case later we do have a parser for it ), and then put the best parser for it. Here Turtle should be right after.